### PR TITLE
Implemented traceListener filters in NLogTraceListener 

### PR DESCRIPTION
**Pull request by thomkinson**
Implementation of filters within the NLog Trace Listener

### CODE REVIEW

**Comment by thomkinson** [comment]
fixes #799

**Comment by 304NotModified** [comment]
Thanks for this PR.

I'm a bit confused. Why can we do this with the regular filters? (https://github.com/NLog/NLog/wiki/Filters)

**Comment by thomkinson** [comment]
I need to configure everything within the trace listeners. Not all logging
goes to nlog so just to simplify configuration and know what is happening.

**Comment by 304NotModified** [comment]
OK clear.

**Comment by 304NotModified** [comment]
Is the filter check don't needed at `TraceTransfer(...) `?

**Comment by thomkinson** [comment]
Wasn't sure how the wiki updates will work. Hopefully adding this submodule solves it

**Comment by 304NotModified** [comment]
We normally just edit the wiki with the edit button on github (yes it's then visible already) ;)

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "wiki"]
+	path = wiki
+	url = https://github.com/thomkinson/NLog.wiki.git
```
Comment (304NotModified) [inline .gitmodules]: I'm not sure if this works.?

File: src/NLog/NLogTraceListener.cs [hunk 2]
```diff
@@ -303,20 +312,10 @@ public override void TraceEvent(TraceEventCache eventCache, string source, Trace
...
-        /// Writes trace information, a message, a related activity identity and event information to the listener specific output.
-        /// </summary>
-        /// <param name="eventCache">A <see cref="T:System.Diagnostics.TraceEventCache"/> object that contains the current process ID, thread ID, and stack trace information.</param>
-        /// <param name="source">A name used to identify the output, typically the name of the application that generated the trace event.</param>
-        /// <param name="id">A numeric identifier for the event.</param>
-        /// <param name="message">A message to write.</param>
-        /// <param name="relatedActivityId">A <see cref="T:System.Guid"/>  object identifying a related activity.</param>
-        public override void TraceTransfer(TraceEventCache eventCache, string source, int id, string message, Guid relatedActivityId)
-        {
-            this.ProcessLogEventInfo(LogLevel.Debug, source, message, null, id, TraceEventType.Transfer, relatedActivityId);
```
Comment (304NotModified) [inline src/NLog/NLogTraceListener.cs]: TraceTransfer has been removed?  Is this a mistake?

File: wiki [hunk 3]
```diff
@@ -0,0 +1 @@
+Subproject commit 84bae1886265228f00510ec6efcf5c7f116fd9b1
```
Comment (304NotModified) [inline wiki]: Do you know how this works?

File: tests/NLog.UnitTests/NLogTraceListenerTests.cs [hunk 4]
```diff
@@ -287,6 +287,49 @@ public void ForceLogLevelTest()
...
+                <nlog>
+                    <targets><target name='debug' type='Debug' layout='${logger} ${level} ${message} ${event-context:EventID}' /></targets>
+                    <rules>
+                        <logger name='*' minlevel='Trace' writeTo='debug' />
+                    </rules>
+                </nlog>");
+
+            TraceSource ts = CreateTraceSource();
+            Func<TraceEventCache, string, TraceEventType, int, string, object[], object, object[], bool> resultFunc =
+                (cache, source, eventType, id, formatOrMessage, args, data1, data) =>
```
Comment (304NotModified) [inline tests/NLog.UnitTests/NLogTraceListenerTests.cs]: I have to agree that this func has a lot of arguments! As general rule, 7 is the max. Can this be improved?

**Comment by 304NotModified** [comment]
Hi @thomkinson,

Do you have time to check te comments? Otherwise we can't include this in 4.1.

**Comment by thomkinson** [comment]
Been heaps busy, so will get onto it tonight. apologies

**Comment by 304NotModified** [comment]
Note: We will close 4.1 next week.

**Comment by 304NotModified** [comment]
Should we move this to 4.2? 

We really like to release 4.1 soon.

**Comment by 304NotModified** [comment]
bump to 4.2. This one need some more docs before merge.

**Comment by 304NotModified** [comment]
Please create a new PR when you have to fix this.

**Comment by drmcclelland** [comment]
This would be awesome - any update on this @thomkinson?

**Pull request closed without merging**